components/brave_wallet/browser/brave_wallet_utils.cc

```diff
@@ -1189,6 +1189,35 @@ absl::optional<std::string> GetChainId(PrefService* prefs,
   return absl::nullopt;
 }
 
+absl::optional<std::string> GetChainIdByNetworkId(
+    PrefService* prefs,
+    const mojom::CoinType& coin,
+    const std::string& network_id) {
+  if (network_id.empty()) {
+    return "";
+  }
+
+  std::vector<mojom::NetworkInfoPtr> networks = GetAllChains(prefs, coin);
+  absl::optional<std::string> chain_id;
+  for (const auto& network : networks) {
+    std::string id = GetKnownNetworkId(coin, network->chain_id);
+    if (id == network_id) {
+      chain_id = network->chain_id;
```

Comment:
You can just return chain_id here and no need to check `if (!chain_id)` below.

Reply:
fixed (simplified the method: GetChainIdByNetworkId)